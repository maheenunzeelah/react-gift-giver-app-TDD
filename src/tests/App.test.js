import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import App from '../components/app';
import ReactDOM from 'react-dom';
import render from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

const app=shallow(<App />);

it("It renders correctly",()=>{
    expect(toJSON(app)).toMatchSnapshot();
});
it("Initializes gift `state` to empty array",()=>{
    expect(app.state().gifts).toEqual([]);
});
it("adds new gift when clicking `add gift`button",()=>{
    app.find('.btn-add').simulate('click');
});

