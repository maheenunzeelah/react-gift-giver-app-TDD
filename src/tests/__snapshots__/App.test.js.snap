// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
  <button
    className="btn btn-primary btn-add"
    onClick={[Function]}
  >
    Add Gift
  </button>
</div>
`;
